authen/src/app.controller.ts

```diff
@@ -0,0 +1,104 @@
+import { BadRequestException, Body, Controller, Get, Post, Res, Req, UnauthorizedException } from '@nestjs/common';
+import { AppService } from './app.service';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from "@nestjs/jwt";
+import { Response, Request } from 'express';
+
+@Controller()
+export class AppController {
+  constructor(
+    private readonly appService: AppService,
+    private jwtService: JwtService
+    ) {}
+
+  @Get()
+  hello(){
+    return {
+      message : "Hello Taka"
+    }
+  }
+
+  @Post('register')
+  async register(
+    @Body('username') username: string,
+    @Body('tel') tel: string,
+    @Body('password') password: string,
+    @Body('confirmPassword') confirmpassword: string,
+  ){
+    const hashedPassword = await bcrypt.hash(password, 12);
```

Comment:
this might have problem sometime. maybe you can find another hashing library later.